utilities/ipplan-import.py

```diff
@@ -0,0 +1,187 @@
+    return p
+
+
+def add_host(host):
+    """
+    Put your host information in the prefix object.
+    """
+    p = new_prefix()
+    p.prefix = str(host['ipaddr'])
+    p.type = "host"
+    p.description = host['description']
+    p.node = host['fqdn']
+    p.comment = ""
+
+    # Use remaining data from ipplan to populate comment field.
+    if 'additional' in host:
+        p.comment += "{}\n".format(host['additional'])
+
+    if len(host['location']) > 0:
+        p.comment += "Location: {}\n".format(host['location'])
+
+    if len(host['mac']) > 0:
+        p.comment += "MAC: {}".format(host['mac'])
+
+    if len(host['phone']) > 0:
+        p.comment += "Phone: {}".format(host['phone'])
+
+    if len(host['user']) > 0:
+        p.comment += "User: {}".format(host['user'])
+
+    return p
+
+
+if __name__ == '__main__':
+    # read configuration
+    cfg = configparser.ConfigParser()
+    cfg.read(os.path.expanduser('~/.nipaprc'))
+
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--username', help="NIPAP backend username")
+    parser.add_argument('--password', help="NIPAP backend password")
+    parser.add_argument('--host', help="NIPAP backend host")
+    parser.add_argument('--port', help="NIPAP backend port")
+    ipplan = parser.add_argument_group()
+    ipplan.add_argument('--base-file', help="Path to ipplan base csv file",
+                        required=True)
+    ipplan.add_argument('--ipaddr-file', help="Path to ipplan ipaddr csv file",
+                        required=True)
+    ipplan.add_argument('--logfile', help="Path to import error log",
+                        default='/tmp/ipplan-import-errors.log')
+    args = parser.parse_args()
+
+    auth_uri = "%s:%s@" % (args.username or cfg.get('global', 'username'),
+                           args.password or cfg.get('global', 'password'))
+
+    xmlrpc_uri = "http://%(auth_uri)s%(host)s:%(port)s" % {
+            'auth_uri': auth_uri,
+            'host': args.host or cfg.get('global', 'hostname'),
+            'port': args.port or cfg.get('global', 'port')
+            }
+    pynipap.AuthOptions({'authoritative_source': 'nipap'})
+    pynipap.xmlrpc_uri = xmlrpc_uri
+
+    networks = get_networks(args.base_file, args.ipaddr_file)
+
+    # Collect all prefixes and hosts that couldn't be saved to NIPAP.
+    log = open(args.logfile, 'a')
+
+    for network in networks:
+        p = add_prefix(network)
+        try:
+            p.save()
+        except Exception as exc:
+            log.write("ERROR: {}\n".format(exc))
+            log.write("INFO: prefix: {}, type: {}, comment: {}\n".format(
+                p.prefix, p.type, p.description))
+
+        for host in network['hosts']:
+            p = add_host(host)
+            try:
+                p.save()
+            except Exception as exc:
+                log.write("ERROR: {}\n".format(exc))
+                log.write("INFO: host: {}, type: {}, node: {}, desc: {} \
```

Comment:
Sometimes I think Hound is over-zealous. I would recommend that you not break the text field of this line. Having the entire text field on one line makes it easier to grep for.